### PR TITLE
Adds DupFinder support

### PR DESCRIPTION
**Pull request by yvschmid**
#350

### CODE REVIEW

File: src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+            Run(settings, GetArgument(settings, files));
+        }
+
+        /// <summary>
+        /// Analyses all files matching the specified pattern using the specified settings.
+        /// </summary>
+        /// <param name="pattern">The pattern.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(string pattern, DupFinderSettings settings)
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs]: I think this method could be moved to the alias. This way we would have only one entry method to the tool.

File: src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            var sourceFiles = _globber.Match(pattern).OfType<FilePath>();
+
+            Run(sourceFiles, settings);
+        }
+
+        /// <summary>
+        /// Runs ReSharper's DupFinder using the provided config file.
+        /// </summary>
+        /// <param name="configFile">The config file.</param>
+        public void RunFromConfig(FilePath configFile)
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs]: I think this method could be moved to the alias. This way we would have only one entry method to the tool.

**Comment by patriksvensson** [comment]
Looks good :+1:

Left a comment about a minor thing, but the change is up to you.

**Comment by yvschmid** [comment]
I've moved the `Run(string, DupFinderSettings method)` to the alias. But I couldn't think of a nice solution for the `RunFromConfig(FilePath)` one. Any ideas?

**Comment by patriksvensson** [comment]
@yvschmid I think that `RunFromConfig` is OK to leave as it is :smile:

**Comment by yvschmid** [comment]
rebase into one commit?

**Comment by patriksvensson** [comment]
@yvschmid Yes please :)

**Comment by patriksvensson** [comment]
Would be great if you could rebase against develop. Just merged your other PR :smile:

**Comment by patriksvensson** [comment]
@yvschmid Would be great if you could rebase against develop. Just merged your other PR :smile:

**Comment by yvschmid** [comment]
@patriksvensson  Done

**Comment by patriksvensson** [comment]
Merged :+1:

**Pull request merged**